[2025-01-05T14:36:19.736+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2025-01-05T14:36:18.444000+00:00 [queued]>
[2025-01-05T14:36:19.746+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2025-01-05T14:36:18.444000+00:00 [queued]>
[2025-01-05T14:36:19.746+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 3
[2025-01-05T14:36:19.764+0000] {taskinstance.py:1327} INFO - Executing <Task(SparkSubmitOperator): submit_spark_job> on 2025-01-05 14:36:18.444000+00:00
[2025-01-05T14:36:19.770+0000] {standard_task_runner.py:57} INFO - Started process 77 to run task
[2025-01-05T14:36:19.774+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_kafka_to_postgres_dag', 'submit_spark_job', 'manual__2025-01-05T14:36:18.444000+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/teams_kafka_to_spark.py', '--cfg-path', '/tmp/tmp61jpbrmb']
[2025-01-05T14:36:19.776+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask submit_spark_job
[2025-01-05T14:36:19.832+0000] {task_command.py:410} INFO - Running <TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2025-01-05T14:36:18.444000+00:00 [running]> on host 32538e397132
[2025-01-05T14:36:20.297+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_kafka_to_postgres_dag' AIRFLOW_CTX_TASK_ID='submit_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-01-05T14:36:18.444000+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-05T14:36:18.444000+00:00'
[2025-01-05T14:36:20.306+0000] {base.py:73} INFO - Using connection ID 'my_spark_conn' for task execution.
[2025-01-05T14:36:20.308+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=log4j.properties --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgres --verbose --queue root.default /usr/local/***/include/scripts/kafka_to_postgres_teams.py
[2025-01-05T14:36:20.327+0000] {spark_submit.py:490} INFO - /usr/bin/env: ‘bash’: No such file or directory
[2025-01-05T14:36:20.338+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=log4j.properties --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgres --verbose --queue root.default /usr/local/***/include/scripts/kafka_to_postgres_teams.py. Error code is: 127.
[2025-01-05T14:36:20.344+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_kafka_to_postgres_dag, task_id=submit_spark_job, execution_date=20250105T143618, start_date=20250105T143619, end_date=20250105T143620
[2025-01-05T14:36:20.364+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 6 for task submit_spark_job (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=log4j.properties --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgres --verbose --queue root.default /usr/local/***/include/scripts/kafka_to_postgres_teams.py. Error code is: 127.; 77)
[2025-01-05T14:36:20.411+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2025-01-05T14:36:20.441+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
